semcore/date-picker/src/components/InputTrigger.jsx

```diff
@@ -0,0 +1,520 @@
+      this.fromRef.current.focus();
+      setTimeout(() => {
+        this.fromRef.current.setSelectionRange(value.length, value.length);
+      }, 0);
+    }
+    if (
+      event.code === 'ArrowLeft' &&
+      this.toRef.current.selectionStart === 0 &&
+      this.toRef.current.selectionEnd === 0
+    ) {
+      const value = this.fromRef.current.value;
+      this.fromRef.current.focus();
+      setTimeout(() => {
+        this.fromRef.current.setSelectionRange(value.length, value.length);
+      }, 0);
+    }
+  };
+
+  getFromMaskedInputProps() {
+    const { value, locale, onDisplayedPeriodChange } = this.asProps;
+
+    return {
+      ref: this.fromRef,
+      date: value?.[0],
+      onDateChange: this.handleFromChange,
+      onKeyDown: this.handleFromKeydown,
+      locale,
+      flex: 1,
+      onDisplayedPeriodChange,
+    };
+  }
+  getToMaskedInputProps() {
+    const { value, locale, onDisplayedPeriodChange } = this.asProps;
+
+    return {
+      ref: this.toRef,
+      date: value?.[1],
+      onDateChange: this.handleToChange,
+      onKeyDown: this.handleToKeydown,
+      locale,
+      flex: 1,
+      onDisplayedPeriodChange,
+    };
+  }
+
+  render() {
+    const SDateRange = Root;
+    const { Children, styles, w } = this.asProps;
+
+    return sstyled(styles)(
+      <SDateRange render={InputMask} __excludeProps={['onChange', 'value']} w={w}>
+        <Children />
+      </SDateRange>,
+    );
+  }
+}
+
+const FromMaskedInput = (props) => {
+  const SFromMaskedInput = Root;
+
+  return sstyled(props.styles)(<SFromMaskedInput labelPrefix="from date" render={MaskedInput} />);
+};
+
+const ToMaskedInput = (props) => {
+  const SToMaskedInput = Root;
+
+  return sstyled(props.styles)(<SToMaskedInput labelPrefix="to date" render={MaskedInput} />);
+};
+
+const Indicator = (props) => {
+  const SIndicator = Root;
+
+  return sstyled(props.styles)(
+    <SIndicator render={InputMask.Addon} tag={Calendar} aria-hidden="true" tabIndex={-1} />,
+  );
+};
+
+const RangeSep = (props) => {
+  const SRangeSep = Root;
+
+  return sstyled(props.styles)(
+    <SRangeSep
+      render={InputMask.Addon}
+      tag={Flex}
+      alignItems="center"
+      justifyContent="center"
+      flex="0"
+      mr={1}
+    >
+      -
+    </SRangeSep>,
+  );
+};
+
+const MaskedInput = ({
+  date: outerValue,
+  onDateChange,
+  onDisplayedPeriodChange,
+  locale,
+  styles,
+  parts: allowedParts = defaultAllowedParts,
+  disabledDates,
+  labelPrefix = 'Date',
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  __excludeProps,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  Root,
+  ...otherProps
+}) => {
+  const { sep, order } = React.useMemo(() => {
+    const exampleDate = new Date(2000, 4, 29);
+    const options = { year: 'numeric', month: '2-digit', day: '2-digit' };
+    const dateTimeFormat = new Intl.DateTimeFormat(locale, options);
+
+    let sep = undefined;
```

Comment:
what does "sep" mean?

Reply:
It is common shortening for the separator word. Like the one exported from std lib `path` package